### ISSUE TITLE
AesEncryptionHelper throws ArgumentException for empty data arrays

### ISSUE THREAD

**Issue by cremor** [UglyToad/PdfPig#409]
I found a PDF that can not be opened by the `PdfDocument.Open()` method. If I try it I get the following exception:

```
UglyToad.PdfPig.Exceptions.PdfDocumentEncryptedException: The document was encrypted and decryption of a token failed. Token was: <Author, (encrypted_data)>, <CreationDate, (encrypted_data)>, <Creator, (encrypted_data)>, <Keywords, ()>, <ModDate, (encrypted_data)>, <Producer, (encrypted_data)>, <RZL_META, (encrypted_data
   at UglyToad.PdfPig.Encryption.EncryptionHandler.Decrypt(IndirectReference reference, IToken token)
   at UglyToad.PdfPig.Tokenization.Scanner.PdfTokenScanner.MoveNext()
   at UglyToad.PdfPig.Tokenization.Scanner.PdfTokenScanner.Get(IndirectReference reference)
   at UglyToad.PdfPig.Parser.Parts.DirectObjectFinder.Get[T](IndirectReference reference, IPdfTokenScanner scanner)
   at UglyToad.PdfPig.Parser.DocumentInformationFactory.Create(IPdfTokenScanner pdfTokenScanner, TrailerDictionary trailer)
   at UglyToad.PdfPig.Parser.PdfDocumentFactory.OpenDocument(IInputBytes inputBytes, ISeekableTokenScanner scanner, ILog log, Boolean isLenientParsing, IReadOnlyList`1 passwords, Boolean clipPaths)
   at UglyToad.PdfPig.Parser.PdfDocumentFactory.Open(IInputBytes inputBytes, ParsingOptions options)
   at UglyToad.PdfPig.Parser.PdfDocumentFactory.Open(String filename, ParsingOptions options)
   at UglyToad.PdfPig.PdfDocument.Open(String filePath, ParsingOptions options)
```

Inner Exception:
```
ArgumentException: Source array was not long enough. Check srcIndex and length, and the array's lower bounds.
   at System.Array.Copy(Array sourceArray, Int32 sourceIndex, Array destinationArray, Int32 destinationIndex, Int32 length, Boolean reliable)
   at System.Array.Copy(Array sourceArray, Array destinationArray, Int32 length)
   at UglyToad.PdfPig.Encryption.AesEncryptionHelper.Decrypt(Byte[] data, Byte[] finalKey)
   at UglyToad.PdfPig.Encryption.EncryptionHandler.DecryptData(Byte[] data, IndirectReference reference)
   at UglyToad.PdfPig.Encryption.EncryptionHandler.DecryptInternal(IndirectReference reference, IToken token)
   at UglyToad.PdfPig.Encryption.EncryptionHandler.DecryptInternal(IndirectReference reference, IToken token)
   at UglyToad.PdfPig.Encryption.EncryptionHandler.Decrypt(IndirectReference reference, IToken token)
```

The inner exceptions shows the actual problem - a call to `Array.Copy()` with an empty source array.

The fix seems simple. Add a check for empty arrays in `AesEncryptionHelper.Decrypt()` - or maybe already in `EncryptionHandler.DecryptData()` - and just return the empty array in that case.
I tried this fix in a locally modified version of PdfPig and with it I could indeed open the PDF file.

Data arrays which are not empty but smaller than 16 bytes will still fail (because that's what the `Array.Copy()` call uses as length), but I don't know how those should be handled. Or if this is even a valid case. If it isn't, then it's fine that an exception is still thrown there.

**Comment by EliotJones** [comment]
@cremor thanks I'll try that fix in the next couple of days, can you see what happens with < 16 arrays if you call copy with length as `Math.Max(16, source.Length)` or whatever the source array is called. I'm not sure if zero padding the array will be fine for those and don't have any documents for reference.

**Comment by EliotJones** [comment]
@cremor yes sorry Min, I always get them reversed. But if there aren't any documents with < 16 arrays then don't worry, iirc the key must be 16 bytes exactly.

**Comment by EliotJones** [comment]
@cremor there should be a pre-release NuGet package containing this fix released tonight at midnight UTC.

**Comment by cremor** [comment]
I can confirm that this is fixed in version 0.1.6-alpha-20220113-5b66e, thanks!
